substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -1289,16 +1402,15 @@ where
 		// transactions with those hashes from the retracted blocks.
 		let mut pruned_log = HashSet::<ExtrinsicHash<ChainApi>>::new();
 
-		future::join_all(
-			tree_route
-				.enacted()
-				.iter()
-				.map(|h| crate::prune_known_txs_for_block(h, &*api, &view.pool)),
-		)
+		future::join_all(tree_route.enacted().iter().map(|hn| {
+			let api = api.clone();
+			async move { (hn, crate::prune_known_txs_for_block(hn, &*api, &view.pool).await) }
+		}))
 		.await
 		.into_iter()
-		.for_each(|enacted_log| {
-			pruned_log.extend(enacted_log);
+		.for_each(|(key, enacted_log)| {
+			pruned_log.extend(enacted_log.clone());
+			self.included_transacations.lock().insert(key.clone(), enacted_log);
```

Comment:
Just for the mental model: This `included_transactions` cache is:
- only updated with enacted blocks when the best block switches between forks
- is cleaned up on finality stall and when we see a new finalized block

Reply:
It is updated when `BestBlock` is notified, even if there is progress on the same fork.